File: Frontend/bookmyplay/src/services/venueService.js
import api from "./api";

export const getAllVenues = () => {
    return api.get("/venues");
};

export const searchVenues = (params) => {
    return api.get("/venues/search", { params });
};


export const getVenueById = (id) => {
    return api.get(`/venues/${id}`);
};

export const getVenuesByVendor = (vendorId) => {
    return api.get(`/venues/vendor/${vendorId}`);
};
export const addVenue = (venueData, files, coverIndex) => {
    const formData = new FormData();
    formData.append(
        "venue",
        new Blob([JSON.stringify(venueData)], { type: "application/json" })
    );
    files.forEach((file) => {
        formData.append("images", file);
    });
    formData.append("coverIndex", coverIndex);

    return api.post("/venues", formData, {
        headers: {
            "Content-Type": "multipart/form-data",
        },
    });
};

export const updateVenue = (id, venue) => {
    return api.put(`/venues/${id}`, venue);
};

export const deleteVenue = (id) => {
    return api.delete(`/venues/${id}`);
};

File: Backend/bookmyplay-api/src/main/java/com/bookmyplay/service/VenueService.java
package com.bookmyplay.service;

import com.bookmyplay.dto.AddVenueRequest;
import com.bookmyplay.entity.Venue;
import java.util.List;

import org.springframework.web.multipart.MultipartFile;

public interface VenueService {

    String addVenue(AddVenueRequest request, MultipartFile[] images, int coverIndex);

    List<Venue> getAllVenues();

    List<Venue> getVendorVenues(Long vendorId);

    Venue getVenueById(Long id);

    org.springframework.data.domain.Page<Venue> searchVenues(com.bookmyplay.dto.VenueSearchDTO searchDTO, org.springframework.data.domain.Pageable pageable);

    String updateVenue(Long id, AddVenueRequest request);

    String deleteVenue(Long id);
}